src/caterpillar/filemanager/gcs.go

```diff
@@ -1,29 +1,94 @@
 package filemanager
 
 import (
-	"appengine"
-	"appengine/file"
+	"fmt"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/file"
+	"google.golang.org/appengine/log"
+	"google.golang.org/cloud"
+	"google.golang.org/cloud/storage"
 )
 
+// image struct holds information needed to run the various image functions.
+type uploadImage struct {
+	c   context.Context
+	w   http.ResponseWriter
+	ctx context.Context
+	// cleanUp is a list of filenames that need cleaning up at the end of the image.
+	cleanUp []string
+	// failed indicates that one or more of the image steps failed.
+	failed bool
+}
+
+func (ui *uploadImage) errorf(format string, args ...interface{}) {
+	ui.failed = true
+	log.Errorf(ui.c, format, args...)
+}
+
 // アップロードされた画像をGCSに保存します。
-func Store(c appengine.Context, data []byte, filename, mimeType, bucketName string) (absFilename string, err error) {
+func Store(c context.Context, w http.ResponseWriter, data []byte, fileName, bucketName string) (absFilename string, err error) {
+
+	if bucketName == "" {
+		var err error
+		if bucketName, err = file.DefaultBucketName(c); err != nil {
+			log.Errorf(c, "failed to get default GCS bucket name: %v", err)
+			return "", err
+		}
+	}
+
+	hc := &http.Client{
+		Transport: &oauth2.Transport{
+			Source: google.AppEngineTokenSource(c, storage.ScopeFullControl),
+		},
+	}
+
+	ctx := cloud.NewContext(appengine.AppID(c), hc)
 
-	opts := &file.CreateOptions{
-		MIMEType:   mimeType,
-		BucketName: bucketName,
+	ui := &uploadImage{
+		c:   c,
+		w:   w,
+		ctx: ctx,
 	}
-	wc, absFilename, err := file.Create(c, filename, opts)
+
+	absFilename, err = ui.createFile(bucketName, fileName, data)
 	if err != nil {
-		c.Errorf("gcs.go:23")
+		fmt.Fprintf(ui.w, "\nUpload failed.\n")
 		return "", err
+	} else {
+		fmt.Fprintf(ui.w, "\nUpload succeeded.\n")
+		return absFilename, nil
 	}
-	defer wc.Close()
+}
 
-	_, err = wc.Write(data)
-	if err != nil {
-		c.Errorf("gcs.go:30")
+// createFile creates a file in Google Cloud Storage.
+func (ui *uploadImage) createFile(bucketName string, fileName string, data []byte) (absFilename string, err error) {
+	fmt.Fprintf(ui.w, "Creating file /%v/%v\n", bucketName, fileName)
+
+	wc := storage.NewWriter(ui.ctx, bucketName, fileName)
+	wc.ContentType = "text/plain"
```

Comment:
files to be uploaded should be image.
old code saved mime-type.
please do the same.